zipline/pipeline/engine.py

```diff
@@ -330,6 +329,11 @@ def run_chunked_pipeline(self, pipeline, start_date, end_date, chunksize):
         )
         chunks = [self.run_pipeline(pipeline, s, e) for s, e in ranges]
 
+        if len(chunks) == 1:
+            # OPTIMIZATION: Don't make an extra copy in `categorical_df_concat`
+            # if we don't have to.
+            return chunks[0]
+
         return categorical_df_concat(chunks, inplace=True)
```

Comment:
The `inplace` parameter implies that a copy is going to be made, and it seems confusing to me to have that not always be true if `chunks` is of length 1.